### PR TITLE
Make ConstantPool(DataInput) constructor public

### PR DESCRIPTION
**Pull request by Maccimo**
Enable manual constant pool loading from external sources.

### CODE REVIEW

File: src/main/java/org/apache/bcel/classfile/ConstantPool.java [hunk 1]
```diff
@@ -57,7 +57,7 @@ public ConstantPool(Constant[] constant_pool) {
      * @throws IOException
      * @throws ClassFormatException
      */
-    ConstantPool(DataInput input) throws IOException, ClassFormatException {
+    public ConstantPool(DataInput input) throws IOException, ClassFormatException {
```
Comment (silviu-burcea) [inline src/main/java/org/apache/bcel/classfile/ConstantPool.java]: You don't have to make it public. Simply move the test to the right package

Comment (Maccimo) [inline src/main/java/org/apache/bcel/classfile/ConstantPool.java]: The reason ConstantPool() should be public is explained here: https://issues.apache.org/jira/browse/BCEL-180

<blockquote>
In some cases such as JDI API we have the class constant pool content only in form of separate byte array. Currently there is no way to load constant pool from byte array.
<br /><br />
The proposed patch enable loading ConstantPool from custom DataInput stream by making ConstantPool(DataInput) constructor public.
</blockquote>

**Comment by garydgregory** [comment]
In BCEL 6.0. the following constructors are public:

- org.apache.bcel.classfile.ConstantPool.ConstantPool(Constant[])
- org.apache.bcel.classfile.ConstantPool.ConstantPool(DataInput)

**Comment by garydgregory** [comment]
Already done.

**Pull request closed without merging**